aws-android-sdk-s3/build.gradle

```diff
@@ -5,7 +5,7 @@ android {
     compileSdkVersion 29
 
     defaultConfig {
-        minSdkVersion 9
+        minSdkVersion 19
```

Comment:
Note that try-with-resources statement requires API level 19.

Reply:
That's interesting. I think we're supposed to support API 16 and up, so we may have to go without the try-with-resources. Let's discuss with the rest of the team to make sure.

Reply:
> Note that try-with-resources statement requires API level 19.

I thought this at first as well.  However, it should actually work as long as you are using gradle plugin 3.0.0 or above which we are.  It includes a tool called `desugar`, which adds backwards compatibility for try-with-resources for all Android API versions.  See the reference about halfway down this page about it: https://developer.android.com/studio/write/java8-support